src/index.js

```diff
@@ -1,39 +1,41 @@
-require("./style.scss");
+import "./style.scss";
+import fullpage from "fullpage.js";
 
-new fullpage("#fullpage", {
-  responsiveWidth: 960,
-  onLeave: function(currentPage, nextPage, direction) {
-    const index = currentPage.index;
-    const nextIndex = nextPage.index;
+(function() {
+  new fullpage("#fullpage", {
+    licenseKey: null,
+    responsiveWidth: 960,
+    onLeave: function(currentPage, nextPage, direction) {
+      const index = currentPage.index;
+      const nextIndex = nextPage.index;
 
-    const updateAnimation = (section, action) => {
-      document.querySelectorAll(`#${section} .imageBox`).forEach(function(e) {
-        if (action === "add") {
-          e.classList.add("imageBox--spin");
-        } else if (action === "remove") {
-          e.classList.remove("imageBox--spin");
-        }
-      });
-      document.querySelectorAll(`#${section} .textBox`).forEach(function(e) {
-        if (action === "add") {
-          e.classList.add("textBox--fade");
-        } else if (action === "remove") {
-          e.classList.remove("textBox--fade");
-        }
-      });
-    };
+      const updateAnimation = (section, action) => {
+        document.querySelectorAll(`#${section} .imageBox`).forEach(function(e) {
+          if (action === "add") {
+            e.classList.add("imageBox--spin");
+          } else if (action === "remove") {
+            e.classList.remove("imageBox--spin");
+          }
+        });
+        document.querySelectorAll(`#${section} .textBox`).forEach(function(e) {
+          if (action === "add") {
+            e.classList.add("textBox--fade");
+          } else if (action === "remove") {
+            e.classList.remove("textBox--fade");
+          }
+        });
+      };
 
-    if (direction === "down") {
-      updateAnimation(`section${index}`, "remove");
-      updateAnimation(`section${index + 1}`, "add");
-    } else if (direction === "up") {
-      updateAnimation(`section${index - 1}`, "add");
-      updateAnimation(`section${index}`, "remove");
+      if (direction === "down") {
+        updateAnimation(`section${index}`, "remove");
+        updateAnimation(`section${index + 1}`, "add");
+      } else if (direction === "up") {
+        updateAnimation(`section${index - 1}`, "add");
+        updateAnimation(`section${index}`, "remove");
+      }
     }
-  }
-});
+  });
```

Comment:
I've commented this in another pull request. not sure why you need to have this anonymous function as wrapper.

Reply:
It has the same effect with `$(document).ready(...));
Make sure js run after DOM ready, so that it won't miss any node while it is not mounted yet.

Reply:
I have never heard of this trick. It doesn't make sense to me that an anonymous function can be the trick as the `DOMContentLoaded` event. 

It actually doesn't wait until DOMContentLoaded.
https://jsbin.com/koxefix/3/edit?html,console,output

```
      (function() {
        console.log(new Date().getTime(), 'executed');
      })();
      document.addEventListener('DOMContentLoaded', function () {
        console.log(new Date().getTime(), 'DOM Ready');
      });
```

Reply:
I used this way for quite a long time and never think it is wrong.
But I after read more responses from [stackoverflow,](https://stackoverflow.com/questions/9899372/pure-javascript-equivalent-of-jquerys-ready-how-to-call-a-function-when-t) I found the accepted was not 100% right.

I'm going to fix this.

Reply:
Just keep questioning. Proving your thoughts by making prototypes. This is totally doesn't make for me and I wrote a prototype to prove it.

`$(() => {})` is different with yours. It's basically `DOMContentLoaded`

The script position does matter. If you place just before the end of the body tag, you don't need to worry domready issues. Instead, you need to do domready checking if you place your JavaScript in `</head>`.

Anonymous function is good for having scope and avoid the global variables pollution while we don't use webpack bundler in the past. The way you made is very common for this purpose. After using webpack, you don't need to do this manually.

However, it still has nothing to do with the domready anyway.